albumentations/augmentations/functional.py

```diff
@@ -876,16 +876,42 @@ def gauss_noise(image, gauss):
     return image + gauss
 
 
-@clipped
-def brightness_contrast_adjust(img, alpha=1, beta=0):
-    img = img.astype('float32')
+def _brightness_contrast_adjust_non_unit(img, alpha=1, beta=0):
+    img = img.copy()
+
     if alpha != 1:
         img *= alpha
     if beta != 0:
         img += beta * np.mean(img)
     return img
 
 
+@preserve_shape
+def _brightness_contrast_adjust_uint(img, alpha=1, beta=0):
+    dtype = np.dtype('uint8')
+
+    max_value = MAX_VALUES_BY_DTYPE[dtype]
+
+    lut = np.arange(0, max_value + 1).astype('float32')
+
+    if alpha != 1:
+        lut *= alpha
+    if beta != 0:
+        lut += beta * np.mean(img)
```

Comment:
why do you think that 
```
    if alpha != 1:
        lut *= alpha
    if beta != 0:
        lut += beta * np.mean(img)
```
and applying two actions after it
is faster then
```
    if alpha != 1:
        img *= alpha
    if beta != 0:
        img += beta * np.mean(img)
```

Reply:
Iw ould belive that power of the 1d array [0, 255] + look up in the table is faster than of the 2d array (N, N, 3)

before on 200 images, using our benchmark
```
Brightness                     56
Contrast                       60
BrightnessContrast             51
```

With the speedup:

```
Brightness                    168
Contrast                      363
BrightnessContrast            172

```